pkgs/top-level/all-packages.nix

```diff
@@ -12236,6 +12246,14 @@ in
   conkeror-unwrapped = callPackage ../applications/networking/browsers/conkeror { };
   conkeror = self.wrapFirefox conkeror-unwrapped { };
 
+  cpp_ethereum = callPackage ../applications/misc/ethereum/cpp {
```

Comment:
The package builds fine with `beignet`, so one may use it as a GPU default since it is under a free license. There is a way to disable the OpenCL part at all if you add a flag `-DETHASHCL=0` to `cmake`.